apps/glambda/task/glambdatask.py

```
@@ -0,0 +1,272 @@
+                                          resource_size, root_path,
+                                          total_tasks)
+        self.method = task_definition.options.method
+        self.args = task_definition.options.args
+        self.verification_metadata = task_definition.options.verification
+        self.verification_type = self.verification_metadata['type']
+        self.dir_manager = dir_manager
+        self.outputs = [
+            os.path.join(
+                dir_manager.get_task_output_dir(task_definition.task_id),
+                output)
+            for output in task_definition.options.outputs
+        ]
+
+    def _get_subtask_data(self) -> Dict[str, Any]:
+        return {
+            'method': self.method,
+            'args': self.args,
+            'content_type': None,
+            'entrypoint': 'python3 /golem/scripts/job.py'
+        }
+
+    def query_extra_data(self, perf_index: float,
+                         node_id: Optional[str] = None,
+                         node_name: Optional[str] = None) -> Task.ExtraData:
+        start_task = self._get_next_task()
+
+        extra_data = self._get_subtask_data()
+
+        ctd = self._new_compute_task_def(
+            subtask_id=self.create_subtask_id(),
+            extra_data=extra_data,
+            perf_index=perf_index
+        )
+
+        subtask_id = ctd['subtask_id']
+
+        logger.debug(
+            'Created new subtask for task. '
+            'task_id=%s, subtask_id=%s, node_id=%s',
+            self.header.task_id,
+            subtask_id,
+            (node_id or '')
+        )
+
+        self.subtasks_given[subtask_id] = {
+            'subtask_id': subtask_id,
+            'subtask_data': extra_data,
+        }
+        self.subtasks_given[subtask_id]['subtask_id'] = subtask_id
+        self.subtasks_given[subtask_id]['status'] = SubtaskStatus.starting
+        self.subtasks_given[subtask_id]['start_task'] = start_task
```

Perhaps unlikely, but still it is possible that `start_task` will be `None`. I'm not sure how well `CoreTask` will handle this :thinking: Perhaps we should throw an exception if we poll for more than there is? At the very least, the message in the error logs should then be more readable :man_shrugging:

---

This will be changed in get_next_task refactoring effort.